### DISCUSSION TITLE
Awesome library for R ml workflows [general]

### DISCUSSION THREAD

**Discussion by bharatsikka** [ropensci/targets#1165] [General]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

Hey team, 

We love targets and have been working on it since a while, its connectivity with SLURM is amazing for us and clustermq has made our work much faster. We also love the DAGs, dashboards and it debugging support. Its been really useful for our ml workflows and we are also looking to find a similar solution in Python as well. Would you be able to suggest the best alternative for targets in python? If we can replicate the same in python it would again be amazing for us as we've already gotten too used to targets in R :p

Looking forward to hearing from you.

Regards,
Sikka

**Comment by ginolhac** [comment]
Hello,
`target` is an absolute gem, fully agreed. An "equivalent" in Python is https://snakemake.github.io/.
For bioinformatic there are many [wrappers](https://snakemake-wrappers.readthedocs.io) which are nice.
However, some of amazing feature of `targets` are absent. The debugging part, the ease to run small parts, the interactivity, the building of dependencies from start to end (`snakemake` does the reverse) are stuff I miss deeply when I use `snakemake`.
Of note, it also works great with `slurm` with pretty advanced features.

Last, even if I don't use it, https://nextflow.io/ is also a very popular workflow manager.

hope this helps

**Reply by bharatsikka** [reply]
We tried snakemake and you are absolutely right interactivity, debugging and dependencies were difficult for us. We also looked at metaflow (but it doesnt work along with slurm), we will evaluate nextflow as well. Thanks for the response

**Comment by b-rodrigues** [comment]
I was wondering these past weeks how easy (and useful) it would be to use targets but with Python functions that you’d call using reticulate. No idea if such an approach is viable though.

**Reply by nviets** [reply]
Speaking of reproducible workflows with reticulate and targets, are there any examples of of such a setup with [rix](https://github.com/b-rodrigues/rix)? Nix is the natural way to set up something like that.

**Comment by wlandau** [comment]
Thanks for the kind words! I will add that I have heard nice things about [Prefect](https://www.prefect.io/), and there are other Python-based pipeline tools listed at https://github.com/pditommaso/awesome-pipeline.

**Reply by bharatsikka** [reply]
Thank for your responding, we are evaluating Prefect, hope to update you soon on our learnings.
